src/GenerateAdmin.php

```diff
@@ -136,18 +139,11 @@ protected function getOptions() {
         ];
     }
 
-    protected function hasSpatieLaravelPermissions() {
-        $composerFile = base_path('composer.json');
-        try {
-            if($composer = $this->files->get($composerFile)) {
-                $composerContent = json_decode($composer, JSON_OBJECT_AS_ARRAY);
-                if(isset($composerContent['require']['spatie/laravel-permission'])) {
-                    return true;
-                }
-            }
-        } catch (FileNotFoundException $e) {
-            return false;
+    protected function shouldGeneratePermissionsMigration() {
+        if (class_exists('\Brackets\Craftable\CraftableServiceProvider')) {
```

Comment:
I was considering Spatie package vs Craftable. I have choosen Craftable because of one reason - in Craftable we define the role Administrator. Imagine what would have happened if we use Spatie package. Unless there is a role Administrator defined it would break. What do you say?

Reply:
It will not brake, because the migration checks if admin role exists, if not, nothing will happen. And the user is requested to change the migration before migrating.

Reply:
Okay, I agree. But what do we gain by using Spatie - if you are using generator with spatie, then you probably don't have `Administrator` role. So you still don't benefit from this check either. So from this point of view, there's no reason to use Spatie as a reference class.

On the other hand - using spatie is too big coupling for me. Coupling to the 3rd party package. What if the class name changes over time? Craftable is our package, we know exactly when it changes.

To support my view, let's review what Generator does. It actually produces the code depending on four our different packages (that are part of Craftable). So binding this particular feature of generator to the Craftable  totally makes sense to me. I see your point, that `what if someone wants to use generator as a standalone`, but they need to set up almost everything by themselves anyway, so I don't find it useful.